internal/dcache/cluster_manager/utils.go

```diff
@@ -0,0 +1,136 @@
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE
+*/
+
+package clustermanager
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-storage-fuse/v2/common"
+	"github.com/Azure/azure-storage-fuse/v2/internal/dcache"
+)
+
+func IsValidClusterMap(cm dcache.ClusterMap) (bool, string) {
+	// top‐level fields
+	if cm.CreatedAt <= 0 {
+		return false, fmt.Sprintf("ClusterMap:: [%+v] Invalid CreatedAt: %d", cm, cm.CreatedAt)
+	}
+	if cm.LastUpdatedAt <= 0 {
+		return false, fmt.Sprintf("ClusterMap:: [%+v] Invalid LastUpdatedAt: %d", cm, cm.LastUpdatedAt)
+	}
+	if cm.LastUpdatedAt < cm.CreatedAt {
+		return false, fmt.Sprintf("ClusterMap:: [%+v] LastUpdatedAt (%d) < CreatedAt (%d)", cm, cm.LastUpdatedAt, cm.CreatedAt)
+	}
+	if !common.IsValidUUID(cm.LastUpdatedBy) {
+		return false, fmt.Sprintf("ClusterMap:: [%+v] Invalid LastUpdatedBy UUID: %q", cm, cm.LastUpdatedBy)
+	}
+
+	// config
+	cfg := cm.Config
+	if cfg.HeartbeatSeconds <= 0 {
+		return false, fmt.Sprintf("ClusterMap:: [%+v] Invalid Config.HeartbeatSeconds: %d", cm, cfg.HeartbeatSeconds)
+	}
+	if cfg.ClustermapEpoch <= 0 {
+		return false, fmt.Sprintf("ClusterMap:: [%+v] Invalid Config.ClustermapEpoch: %d", cm, cfg.ClustermapEpoch)
+	}
+
+	// RVMap
+
+	shouldReturn, b, s := IsValidRVMap(cm.RVMap)
+	if shouldReturn {
+		return b, s
+	}
+
+	// MVMap
+	for name, mv := range cm.MVMap {
+		if name == "" {
+			return false, fmt.Sprintf("ClusterMap:: [%+v] MVMap: empty key", cm)
+		}
+		switch mv.State {
+		case dcache.StateOnline, dcache.StateOffline:
+		default:
+			return false, fmt.Sprintf("ClusterMap:: [%+v] MVMap[%q]: Invalid mv State: %q", cm, name, mv.State)
+		}
+		if mv.Rvs == nil {
+			return false, fmt.Sprintf("ClusterMap:: [%+v] MVMap[%q]: Rvs is nil", cm, name)
+		}
+		for rvName, state := range mv.Rvs {
+			if rvName == "" {
+				return false, fmt.Sprintf("ClusterMap:: [%+v] MVMap[%q].Rvs: empty key", cm, name)
+			}
+			switch state {
+			case string(dcache.StateOnline), string(dcache.StateOffline):
+			default:
+				return false, fmt.Sprintf("ClusterMap:: [%+v] MVMap[%q].Rvs[%q]: Invalid RV state %q in MV ", cm, name, rvName, state)
+			}
```

Comment:
just checking that rvName is prsent in cm.RVMap should be sufficient as we have already verified that all RVs in cm.RVMap are valid.

but RVs in MV list can have the following additional states

outofsync, syncing